flink-runtime/src/main/java/org/apache/flink/runtime/deployment/ShuffleDeploymentDescriptor.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.deployment;
+
+import org.apache.flink.runtime.io.network.ConnectionID;
+
+import java.io.Serializable;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Deployment descriptor for shuffle specific information.
+ */
+public class ShuffleDeploymentDescriptor implements Serializable {
```

Comment:
I think eventually it needs to be an interface, probably an empty one. This one could stay an implementation for the default shuffle master. Also, special `UnknownShuffleDeploymentDescriptor` could extend the interface.

Reply:
It might need an explicit method of `getConnectionId` in the interface if to do? Because the ICDD might either see `UnknownShuffleDeploymentDescriptor` or `KnownShuffleDeploymentDescriptor` and it should provide the way of getting `ConnectionID` if `LocationType==Remote`.

Reply:
True, for existing netty shuffle, it has to have more methods. Internally, I would suggest, the future NettyShuffleService will cast SDD to KnownNettySDD if it is not an UnknownSDD:
```
interface SDD {  }

enum UnknownSDD implements SDD { INSTANCE; } // special singleton stub

class KnownNettySDD implements SDD { + ProducerResourceId, ProducerConnection, etc }

// later:
class AnyOtherSDD implements SDD { other specific shuffle identification }
```